File: Programming/C++/00/0003 Matrix Addition.cpp
#include<bits/stdc++.h>
using namespace std;
int M,N;
int matrix_01[105][105], matrix_02[105][105], matrix_sum[105][105];

int main ()
{
    // Input row (M) and column (N)
    cin >> M >> N;

    // Input first matrix
    for(int i=1;i<=M;i++)
    {
        for(int j=1;j<=N;j++)
        {
            cin >> matrix_01[i][j];
        }
    }

    // Input second matrix
    // Calculate the sum of 2 matrix
    for(int i=1;i<=M;i++)
    {
        for(int j=1;j<=N;j++)
        {
            cin >> matrix_02[i][j];
            matrix_sum[i][j] = matrix_01[i][j] + matrix_02[i][j];
        }
    }

    // Output the sum of 2 matrix
    for(int i=1;i<=M;i++)
    {
        for(int j=1;j<=N;j++)
        {
            cout << matrix_sum[i][j] << " ";
        }
        cout << "\n";
    }

    // End of the program
    return 0;
}

File: Python/0003 Matrix Addition.py
# Input rows (M) and column (N)
# 3 2 --> ["3", "2"]
data = input().split()
M,N = int(data[0]), int(data[1])

# Create empty matrix
matrix_01 = []
matrix_02 = []

# Input first matrix
# Example input
# 1 2 3
# 3 2 1
# 1 2 3
# matrix_01 = [['1','2','3'],['3','2','1'],['1','2','3']]
for i in range(0,M):
    row = input().split()
    matrix_01.append(row)

# Input second matrix
# Example input
# 1 1 1
# 1 1 1
# 1 1 1
# matrix_01 = [['1','1','1'],['1','1','1'],['1','1','1']]
for i in range(0,M):
    row = input().split()
    matrix_02.append(row)

# Calculate and output a sum of 2 matrix
for i in range(0,M):
    line = ""
    for j in range(0,N):
        line = line + str(int(matrix_01[i][j]) + int(matrix_02[i][j])) + " "
    print(line)